### ISSUE TITLE
Trying this on Redmi Note 11 

### ISSUE THREAD

**Issue by daeiron** [Peter-Easton/android-debug-cable-howto#11]
Hi, 
I have been trying this on my Redmi Note 11. 
I have tried 3 different usb-serial adapters (ft232, cp2102,pl2303). Also changed cable and flipped connection.

Enabled these configs in kernel configuration:
CONFIG_USB_SERIAL=y
CONFIG_USB_SERIAL_CONSOLE=y
CONFIG_USB_SERIAL_CP210X=y
CONFIG_USB_SERIAL_PL2303=y
CONFIG_USB_SERIAL_FTDI_SIO=y

But fastboot for this device doesn't have an enable UART option. So can't do that.
Any idea what else to try? Or where to go from here.

I want to be able to debug a non booting kernel.

**Comment by samueldr** [comment]
**DO NOT**

**Comment by samueldr** [comment]
The way this cable operates is through using SBU *(sideband use)* lanes on the Type-C connection.

These lanes can be connected (or not) to *anything* in *any way*. At worst, you can cause permanent damage to circuitry by using a cable not meant to be used on a particular device.

At best (when they add negotiation and such things) they do nothing.

The only way cables with SBU can be used, like this one, is if the device was made specifically for the cable. The "cereal" cable scheme used by Pixel devices is, as far as anyone knows, only used by pixel devices starting with the Pixel 2.

**Comment by Guilher89** [comment]
> Hi, 
> I have been trying this on my Redmi Note 11. 
> I have tried 3 different usb-serial adapters (ft232, cp2102,pl2303). Also changed cable and flipped connection.
> 
> Enabled these configs in kernel configuration:
> CONFIG_USB_SERIAL=y
> CONFIG_USB_SERIAL_CONSOLE=y
> CONFIG_USB_SERIAL_CP210X=y
> CONFIG_USB_SERIAL_PL2303=y
> CONFIG_USB_SERIAL_FTDI_SIO=y
> 
> But fastboot for this device doesn't have an enable UART option. So can't do that.
> Any idea what else to try? Or where to go from here.
> 
> I want to be able to debug a non booting kernel.
> 

Dear [daeiron],

 I hope this message finds you well.  I'm contacting you for information and guidance on compiling the SPES kernel.  I am interested in understanding the compilation process and its specificities.

 So that I can gain a more comprehensive understanding, I would like to ask you a few specific questions:

 1. **Toolchain Used:** Which toolchain (set of compilation tools) did you use to compile the SPES kernel?  Was it a specific toolchain recommended by the manufacturer or one of your choice?

 2. **Compilation Script:** During the compilation process, did you use any custom kernel compilation script or a manufacturer-supplied script?  If yes, could you share details about this script or its settings?

 3. **Specific Settings:** Were there any special settings or adjustments you needed to make when compiling the SPES kernel, especially related to specific hardware or optimizations?

 4. **Issues or Challenges:** While compiling the kernel, did you face any specific issues or challenges that you would like to highlight?  How did you resolve these issues, if any?

 If you are willing to share your experiences and knowledge about compiling the SPES kernel, I would be extremely grateful.  Your information will be invaluable to my own compiling journey.

 Please know that any details or guidance you can provide will be highly appreciated.  If you prefer to share this information in a more private setting, I am happy to exchange private messages or emails, whichever is most convenient for you.

 Thank you in advance for your help and collaboration.  Your knowledge will be of great benefit.

 Yours sincerely,
 [K0WHITE]

**Comment by shinyquagsire23** [comment]
> The only way cables with SBU can be used, like this one, is if the device was made specifically for the cable. The "cereal" cable scheme used by Pixel devices is, as far as anyone knows, only used by pixel devices starting with the Pixel 2.

necrobump, but fwiw I can actually confirm that this is also the method LG phones used, however there was no way to enable it in software afaict and the entire stack from bootrom->ABL->kernel was set up to not use those pins without ~some fusing or NVRAM-ish bit, I forget exactly. They also had some stuff that checked for debug resistors on the CC pins as well. This was a mid-range phone as well, Stylo 4, so no doubt it extended to the higher end ones. But I successfully forced output at S-EL3 with some regpokes to force UART to multiplex onto the pins.

tl;dr it might be a configuration on a lot of phones, is definitely one on LG phones, just one that's only enabled in userdebug/engineering builds if I had to guess
